release_workspace.js

```diff
@@ -0,0 +1,294 @@
+ * @param {string} newVersion The new version to bump to.
+ * @param {string} npm2FACode The 2FA code for NPM.
+ * @param {string | undefined} tag An optional tag to publish to.
+ * @returns {Promise<void>}
+ */
+async function runYarnRelease(packageDirectory, newVersion, npm2FACode, tag) {
+  return new Promise((resolve, reject) => {
+    const workingDirectory = path.join(process.cwd(), packageDirectory);
+    const tagArg = tag ? `--npm.tag=${tag}` : "";
+    const args = ["release-it", "--ci", `--npm.otp=${npm2FACode}`, tagArg, "--config", ".release-it.json", newVersion];
+    
+    console.log(`Running command: "yarn ${args.join(" ")}"`);
+
+    const yarnReleaseProcess = spawn("yarn", args, { stdio: "inherit", cwd: workingDirectory });
+
+    yarnReleaseProcess.on("close", (code) => {
+      if (code === 0) {
+        resolve();
+      } else {
+        reject(`Process exited with code ${code}`);
+      }
+    });
+
+    yarnReleaseProcess.on("error", (err) => {
+      reject(err);
+    });
+  });
+}
+
+/**
+ * Finds all `package.json`'s which contain the input workspace as a dependency.
+ * Then, updates the dependency to the new version, runs yarn install and
+ * commits the changes.
+ * 
+ * @param {string} workspaceName The name of the workspace to bump dependencies for.
+ * @param {string} newVersion The new version to bump to.
+ * @param {Array<{ dir: string, packageJSON: Record<string, any>}>} allWorkspaces
+ * @param {string | undefined} tag An optional tag to publish to.
+ * @returns {void}
+ */
+function bumpDeps(workspaceName, newVersion, allWorkspaces, tag) {
+  console.log(`Bumping other packages which depend on ${workspaceName}.`);
+  console.log("Checking out main branch.");
+
+  // Separate variable for the branch name, incase it includes a tag.
+  let versionString = newVersion;
+  if (tag) {
+    versionString = `${newVersion}-${tag}`;
+  }
+
+  execSync(`git checkout main`);
+  const newBranchName = `bump-${workspaceName}-to-${versionString}`;
+  console.log(`Checking out new branch: ${newBranchName}`);
+  execSync(`git checkout -b ${newBranchName}`);
+
+  const allWorkspacesWhichDependOn = allWorkspaces.filter(({ packageJSON }) => 
+    Object.keys(packageJSON.dependencies ?? {}).includes(workspaceName)
+  );
+  const allWorkspacesWhichDevDependOn = allWorkspaces.filter(({ packageJSON }) => 
+    Object.keys(packageJSON.devDependencies ?? {}).includes(workspaceName)
+  );
+  const allWorkspacesWhichPeerDependOn = allWorkspaces.filter(({ packageJSON }) =>
+    Object.keys(packageJSON.peerDependencies ?? {}).includes(workspaceName)
+  );
+
+  // For console log, get all workspaces which depend and filter out duplicates.
+  const allWhichDependOn = new Set([
+    ...allWorkspacesWhichDependOn,
+    ...allWorkspacesWhichDevDependOn,
+    ...allWorkspacesWhichPeerDependOn,
+  ].map(({ packageJSON }) => packageJSON.name));
+
+  if (allWhichDependOn.size !== 0) {
```

Comment:
We don't always want to do this though? Only if we ask to bump?

Reply:
This only runs if you pass the `--bump-deps` flag